spinnaker/spinnaker_system/google_http_upsert_server_test.py

```diff
@@ -0,0 +1,195 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# See testable_service/integration_test.py and spinnaker_testing/spinnaker.py
+# for more details.
+#
+# The Http(s) test will be used to (more) thoroughly test the GCP L7 load
+# balancer Upsert and Delete commands, and will eventually be expanded to
+# include server group tests.
+
+# Standard python modules.
+from OpenSSL import crypto
+import copy
+import json
+import sys
+import time
+import uuid
+import google_http_upsert_test as upsert
+
+# citest modules.
+from citest.base import ExecutionContext
+import citest
+import citest.gcp_testing as gcp
+import citest.json_predicate as jp
+import citest.service_testing as st
+
+# Spinnaker modules.
+import spinnaker_testing as sk
+import spinnaker_testing.gate as gate
+
+
+SCOPES = [gcp.COMPUTE_READ_WRITE_SCOPE]
+
+class GoogleHttpLoadBalancerTest(st.AgentTestCase):
+  '''Test fixture for Http LB test.
+  '''
+
+  FIRST_CERT = ''
+  SECOND_CERT = ''
+
+  @classmethod
+  def setUpClass(cls):
+    cls.FIRST_CERT = 'first-cert' + str(uuid.uuid4())
+    cls.SECOND_CERT = 'second-cert' + str(uuid.uuid4())
+    runner = citest.base.TestRunner.global_runner()
+    bindings = runner.bindings
+    scenario = runner.get_shared_data(upsert.GoogleHttpLoadBalancerTestScenario)
+    managed_region = runner.bindings['TEST_GCE_REGION']
+    title = 'Check Quota for {0}'.format(scenario.__class__.__name__)
+
+    verify_results = gcp.verify_quota(
+      title,
+      scenario.gcp_observer,
+      project_quota=upsert.GoogleHttpLoadBalancerTestScenario.MINIMUM_PROJECT_QUOTA,
+      regions=[(managed_region,
+                upsert.GoogleHttpLoadBalancerTestScenario.MINIMUM_REGION_QUOTA)]
+    )
+    if not verify_results:
+      raise RuntimeError('Insufficient Quota: {0}'.format(verify_results))
+
+    key = crypto.PKey()
```

Comment:
move this block plus the ssl_cert declaration into a "make_ssl_cert(name)" function.